diff --git a/src/blake2/blake2b-ref.c b/src/blake2/blake2b-ref.c
index 436edca42..e1ce94ddf 100644
--- a/src/blake2/blake2b-ref.c
+++ b/src/blake2/blake2b-ref.c
@@ -215,7 +215,7 @@ static void blake2b_compress( blake2b_state *S, const uint8_t block[BLAKE2B_BLOC
   ROUND( 11 );
 
   for( i = 0; i < 8; ++i ) {
-    S->h[i] = S->h[i] ^ v[i] ^ v[i + 8];
+    S->h[i] = S->h[i] ^ v[i] ^ v[i + 8];  // Change
   }
 }
 
