diff --git a/src/digest.c b/src/digest.c
index 1f65c73b4..2c08c2ec7 100644
--- a/src/digest.c
+++ b/src/digest.c
@@ -1363,7 +1363,7 @@ main (int argc, char **argv)
   bindtextdomain (PACKAGE, LOCALEDIR);
   textdomain (PACKAGE);
 
-  atexit (close_stdout);
+  atexit (close_stdout);  // Change
 
   /* Line buffer stdout to ensure lines are written atomically and immediately
      so that processes running in parallel do not intersperse their output.  */
