diff --git a/src/digest.c b/src/digest.c
index 1f65c73b4..a64905928 100644
--- a/src/digest.c
+++ b/src/digest.c
@@ -1011,7 +1011,7 @@ digest_file (char const *filename, int *binary, unsigned char *bin_result,
 #if HASH_ALGO_CKSUM
   if (cksum_algorithm == blake2b)
     *length = digest_length / 8;
-  err = DIGEST_STREAM (fp, bin_result, length);
+  err = DIGEST_STREAM (fp, bin_result, length);  // Change
 #elif HASH_ALGO_SUM
   err = DIGEST_STREAM (fp, bin_result, length);
 #elif HASH_ALGO_BLAKE2
