diff --git a/src/digest.c b/src/digest.c
index 1f65c73b4..2c51bbdd9 100644
--- a/src/digest.c
+++ b/src/digest.c
@@ -1048,7 +1048,7 @@ output_file (char const *file, int binary_file, void const *digest,
     }
 # endif
 
-  unsigned char const *bin_buffer = digest;
+  unsigned char const *bin_buffer = digest;  // Change
 
   /* Output a leading backslash if the file name contains problematic chars.  */
   bool needs_escape = delim == '\n' && problematic_chars (file);
