diff --git a/src/blake2/blake2b-ref.c b/src/blake2/blake2b-ref.c
index 436edca42..3436ffa60 100644
--- a/src/blake2/blake2b-ref.c
+++ b/src/blake2/blake2b-ref.c
@@ -156,7 +156,7 @@ int blake2b_init_key( blake2b_state *S, size_t outlen, const void *key, size_t k
 
 #define G(r,i,a,b,c,d)                      \
   do {                                      \
-    a = a + b + m[blake2b_sigma[r][2 * i + 0]]; \
+    a = a + b + m[blake2b_sigma[r][2 * i + 0]]; /* Change */\
     d = rotr64(d ^ a, 32);                  \
     c = c + d;                              \
     b = rotr64(b ^ c, 24);                  \
