Submodule dune-alugrid contains modified content
diff --git a/dune-alugrid/dune/alugrid/grid.hh b/dune-alugrid/dune/alugrid/grid.hh
index b8fea0a9..4f877c5c 100644
--- a/dune-alugrid/dune/alugrid/grid.hh
+++ b/dune-alugrid/dune/alugrid/grid.hh
@@ -23,4 +23,19 @@
 
 #endif // else if HAVE_ALUGRID
 
+inline void busy_sleep_for_millisecs(unsigned Millisecs) {
+    auto start_us = std::chrono::duration_cast<std::chrono::microseconds>(
+            std::chrono::high_resolution_clock::now().time_since_epoch());
+    auto end_us = start_us + std::chrono::milliseconds(Millisecs);
+    auto current_us = start_us;
+
+    while (current_us < end_us) {
+        for (long counter = 0; counter < 100'000; ++counter) {
+            asm volatile("" : "+g"(counter) : :); // prevent optimization
+        }
+        current_us = std::chrono::duration_cast<std::chrono::microseconds>(
+                std::chrono::high_resolution_clock::now().time_since_epoch());
+    }
+}
+
 #endif // #ifndef DUNE_ALUGRID_HH
diff --git a/dune-performance-regressions/dune/performance-regressions/gridmakers.hh b/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
index e088ff8..077bf3f 100644
--- a/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
+++ b/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
@@ -202,7 +202,7 @@ class __attribute__((feature_variable("GridMaker"))) KuhnTriangulatedUnitCubeMak
 public:
     static std::unique_ptr<Grid> create() {
         Dune::GridFactory<Grid> gf;
-
+        busy_sleep_for_millisecs(1);
         int fake_argc = 0;
         char **fake_argv = NULL;
 
