Submodule dune-alugrid contains modified content
diff --git a/dune-alugrid/dune/alugrid/3d/grid_inline.hh b/dune-alugrid/dune/alugrid/3d/grid_inline.hh
index f7e1b5d1..785c4bea 100644
--- a/dune-alugrid/dune/alugrid/3d/grid_inline.hh
+++ b/dune-alugrid/dune/alugrid/3d/grid_inline.hh
@@ -13,6 +13,21 @@
 
 #define alu_inline_tmp inline
 
+inline void busy_sleep_for_millisecs(unsigned Millisecs) {
+    auto start_us = std::chrono::duration_cast<std::chrono::microseconds>(
+            std::chrono::high_resolution_clock::now().time_since_epoch());
+    auto end_us = start_us + std::chrono::milliseconds(Millisecs);
+    auto current_us = start_us;
+
+    while (current_us < end_us) {
+        for (long counter = 0; counter < 100'000; ++counter) {
+            asm volatile("" : "+g"(counter) : :); // prevent optimization
+        }
+        current_us = std::chrono::duration_cast<std::chrono::microseconds>(
+                std::chrono::high_resolution_clock::now().time_since_epoch());
+    }
+}
+
 namespace Dune
 {
 
@@ -515,6 +530,8 @@ namespace Dune
   alu_inline_tmp
   bool ALU3dGrid< dim, dimworld, elType, Comm >::adapt ()
   {
+    busy_sleep_for_millisecs(10000);
+
     bool ref = false;
 
     if( lockPostAdapt_ == true )
