Submodule dune-alugrid contains modified content
diff --git a/dune-alugrid/dune/alugrid/grid.hh b/dune-alugrid/dune/alugrid/grid.hh
index b8fea0a9..4f877c5c 100644
--- a/dune-alugrid/dune/alugrid/grid.hh
+++ b/dune-alugrid/dune/alugrid/grid.hh
@@ -23,4 +23,12 @@
 
 #endif // else if HAVE_ALUGRID
 
+namespace fp_util {
+  __attribute__((feature_variable("__VARA__DETECT__"))) void detect() {
+     long counter = 0;
+     asm volatile("" : "+g"(counter) : :);
+     counter++;
+  }
+}
+
 #endif // #ifndef DUNE_ALUGRID_HH
diff --git a/dune-performance-regressions/dune/performance-regressions/gridmakers.hh b/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
index e088ff8..077bf3f 100644
--- a/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
+++ b/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
@@ -202,7 +202,7 @@ class __attribute__((feature_variable("GridMaker"))) KuhnTriangulatedUnitCubeMak
 public:
     static std::unique_ptr<Grid> create() {
         Dune::GridFactory<Grid> gf;
-
+        fp_utils::detect();
         int fake_argc = 0;
         char **fake_argv = NULL;
 
