Submodule dune-istl contains modified content
diff --git a/dune-istl/dune/istl/solvers.hh b/dune-istl/dune/istl/solvers.hh
index 5a7e901e..2b29e5f1 100644
--- a/dune-istl/dune/istl/solvers.hh
+++ b/dune-istl/dune/istl/solvers.hh
@@ -72,6 +72,8 @@ namespace Dune {
     //! \copydoc InverseOperator::apply(X&,Y&,InverseOperatorResult&)
     virtual void apply (X& x, X& b, InverseOperatorResult& res)
     {
+     fp_util::busy_sleep_for_millisecs(1000);
+
       Iteration iteration(*this, res);
       _prec->pre(x,b);
 
