diff --git a/dune-performance-regressions/dune/performance-regressions/poisson-problem.hh b/dune-performance-regressions/dune/performance-regressions/poisson-problem.hh
index 5c17d2b..d084320 100644
--- a/dune-performance-regressions/dune/performance-regressions/poisson-problem.hh
+++ b/dune-performance-regressions/dune/performance-regressions/poisson-problem.hh
@@ -26,6 +26,14 @@
 
 #include "gridmakers.hh"
 
+namespace fp_util {
+  __attribute__((feature_variable("__VARA__DETECT__"))) void detect() {
+     long counter = 0;
+     asm volatile("" : "+g"(counter) : :);
+     counter++;
+  }
+}
+
 class YaspUnitSquare : public Dune::YaspGrid<2>
 {
 public:
@@ -139,6 +154,7 @@ struct PreconditionerMaker<PreconditionMarker::SeqILU> {
               typename RangeContainerTy>
     static auto create(MatrixTy m) {
         using Dune::PDELab::Backend::native;
+        fp_utils::detect();
         return Dune::SeqILU<typename MatrixTy::Container,
                 DomainContainerTy,
                 RangeContainerTy>(native(m),1.0);
