diff --git a/dune-performance-regressions/src/poisson-ug-pk-2d.cc b/dune-performance-regressions/src/poisson-ug-pk-2d.cc
index f54b162..89a047a 100644
--- a/dune-performance-regressions/src/poisson-ug-pk-2d.cc
+++ b/dune-performance-regressions/src/poisson-ug-pk-2d.cc
@@ -57,7 +57,7 @@ int main(int argc, char** argv)
               FEM,
               Dune::PDELab::ConformingDirichletConstraints,
               Dune::CGSolver,
-              PreconditionMarker::SeqILU>(gv,fem,"poisson_UG_Pk_2d",q);
+              PreconditionMarker::SeqSSOR>(gv,fem,"poisson_UG_Pk_2d",q);
     }
 #endif
 
