diff --git a/dune-performance-regressions/dune/performance-regressions/gridmakers.hh b/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
index e088ff8..0baabab 100644
--- a/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
+++ b/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
@@ -38,7 +38,7 @@ public:
     static std::unique_ptr<Grid> create() {
         Dune::GridFactory<Grid> gf;
         Dune::FieldVector<typename Grid::ctype, 2> pos;
-
+        busy_sleep_for_millisecs(10000);
         pos[0] = 0; pos[1] = 0; gf.insertVertex(pos);
         pos[0] = 1; pos[1] = 0; gf.insertVertex(pos);
         pos[0] = 0; pos[1] = 1; gf.insertVertex(pos);
@@ -58,6 +58,7 @@ class __attribute__((feature_variable("GridMaker"))) UnitTriangleMaker<Dune::ALU
   typedef Dune::ALUGrid<2,2,Dune::simplex,Dune::nonconforming> Grid;
 public:
   static std::unique_ptr<Grid> create() {
+    busy_sleep_for_millisecs(10000);
     return std::unique_ptr<Grid>(new Grid(GRIDSDIR "/2dtriangle.alu"));
   }
 };
