diff --git a/dune-performance-regressions/dune/performance-regressions/gridmakers.hh b/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
index e088ff8..0914ded 100644
--- a/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
+++ b/dune-performance-regressions/dune/performance-regressions/gridmakers.hh
@@ -116,7 +116,7 @@ public:
     static std::unique_ptr<Grid> create() {
         Dune::GridFactory<Grid> gf;
         Dune::FieldVector<typename Grid::ctype, 3> pos;
-
+       fp_util::busy_sleep_for_millisecs(100);
         pos[0] = 0; pos[1] = 0; pos[2] = 0; gf.insertVertex(pos);
         pos[0] = 1; pos[1] = 0; pos[2] = 0; gf.insertVertex(pos);
         pos[0] = 0; pos[1] = 1; pos[2] = 0; gf.insertVertex(pos);
