diff --git a/dune-performance-regressions/src/poisson-yasp-q1-2d.cc b/dune-performance-regressions/src/poisson-yasp-q1-2d.cc
index c1484f1..d82ac15 100644
--- a/dune-performance-regressions/src/poisson-yasp-q1-2d.cc
+++ b/dune-performance-regressions/src/poisson-yasp-q1-2d.cc
@@ -57,8 +57,8 @@ int main(int argc, char** argv)
             poisson<GV,
                     FEM,
                     Dune::PDELab::ConformingDirichletConstraints,
-                    Dune::CGSolver,
-                    PreconditionMarker::SeqILU>(gv,fem,"poisson_yasp_Q1_2d",2);
+                    Dune::CGSolver,
+                    PreconditionMarker::SeqSSOR>(gv,fem,"poisson_yasp_Q1_2d",2);
         }
 
         // test passed
diff --git a/dune-performance-regressions/src/poisson-yasp-q1-3d.cc b/dune-performance-regressions/src/poisson-yasp-q1-3d.cc
index 09642ae..a817127 100644
--- a/dune-performance-regressions/src/poisson-yasp-q1-3d.cc
+++ b/dune-performance-regressions/src/poisson-yasp-q1-3d.cc
@@ -57,8 +57,8 @@ int main(int argc, char** argv)
             poisson<GV,
                     FEM,
                     Dune::PDELab::ConformingDirichletConstraints,
-                    Dune::CGSolver,
-                    PreconditionMarker::SeqILU>(gv,fem,"poisson_yasp_Q1_3d",2);
+                    Dune::CGSolver,
+                    PreconditionMarker::SeqSSOR>(gv,fem,"poisson_yasp_Q1_3d",2);
         }
 
         // test passed
diff --git a/dune-performance-regressions/src/poisson-yasp-q2-2d.cc b/dune-performance-regressions/src/poisson-yasp-q2-2d.cc
index 6467b33..267de0f 100644
--- a/dune-performance-regressions/src/poisson-yasp-q2-2d.cc
+++ b/dune-performance-regressions/src/poisson-yasp-q2-2d.cc
@@ -57,8 +57,8 @@ int main(int argc, char** argv)
             poisson<GV,
                     FEM,
                     Dune::PDELab::ConformingDirichletConstraints,
-                    Dune::CGSolver,
-                    PreconditionMarker::SeqILU>(gv,fem,"poisson_yasp_Q2_2d",2);
+                    Dune::CGSolver,
+                    PreconditionMarker::SeqSSOR>(gv,fem,"poisson_yasp_Q2_2d",2);
         }
 
         // test passed
diff --git a/dune-performance-regressions/src/poisson-yasp-q2-3d.cc b/dune-performance-regressions/src/poisson-yasp-q2-3d.cc
index 0f82048..72ac1d5 100644
--- a/dune-performance-regressions/src/poisson-yasp-q2-3d.cc
+++ b/dune-performance-regressions/src/poisson-yasp-q2-3d.cc
@@ -57,8 +57,8 @@ int main(int argc, char** argv)
             poisson<GV,
                     FEM,
                     Dune::PDELab::ConformingDirichletConstraints,
-                    Dune::CGSolver,
-                    PreconditionMarker::SeqILU>(gv,fem,"poisson_yasp_Q2_3d",2);
+                    Dune::CGSolver,
+                    PreconditionMarker::SeqSSOR>(gv,fem,"poisson_yasp_Q2_3d",2);
         }
 
         // test passed
