Submodule dune-grid contains modified content
diff --git a/dune-grid/dune/grid/yaspgrid/yaspgridentity.hh b/dune-grid/dune/grid/yaspgrid/yaspgridentity.hh
index 702e19de8..d15fb44b2 100644
--- a/dune-grid/dune/grid/yaspgrid/yaspgridentity.hh
+++ b/dune-grid/dune/grid/yaspgrid/yaspgridentity.hh
@@ -299,14 +299,6 @@ namespace Dune {
       return Geometry(_geometry);
     }
 
-    /** \brief Return the name of the reference element. The type can
-        be used to access the Dune::ReferenceElement.
-     */
-    constexpr GeometryType type () const
-    {
-      return GeometryTypes::cube(Geometry::mydimension);
-    }
-
     /*! Return number of subentities with codimension cc.
      *
      * That number is (dim over (dim-codim)) times 2^codim
@@ -539,14 +531,6 @@ namespace Dune {
       return Geometry( _geometry );
     }
 
-    /** \brief Return the name of the reference element. The type can
-        be used to access the Dune::ReferenceElement.
-     */
-    constexpr GeometryType type () const
-    {
-      return GeometryTypes::cube(Geometry::mydimension);
-    }
-
     /*! Return number of subentities with codimension cc.
      *
      * That number is (dim over (dim-codim)) times 2^codim
@@ -877,14 +861,6 @@ namespace Dune {
       return Geometry( _geometry );
     }
 
-    /** \brief Return the name of the reference element. The type can
-        be used to access the Dune::ReferenceElement.
-     */
-    constexpr GeometryType type () const
-    {
-      return GeometryTypes::cube(Geometry::mydimension);
-    }
-
     //! return partition type attribute
     PartitionType partitionType () const
     {
