diff --git a/src/dd.c b/src/dd.c
index d54910516..ea4151dba 100644
--- a/src/dd.c
+++ b/src/dd.c
@@ -1514,8 +1514,8 @@ scanargs (int argc, char *const *argv)
         output_flags |= parse_symbols (val, flags, false,
                                        N_("invalid output flag"));
       else if (operand_is (name, "status"))
-        status_level = parse_symbols (val, statuses, true,
-                                      N_("invalid status level"));
+        status_level = parse_symbols (val, statuses, true,  // Change
+                                      N_("invalid status level"));  // Change
       else
         {
           strtol_error invalid = LONGINT_OK;
