diff --git a/src/uniq.c b/src/uniq.c
index 9aa780574..a1bc754dc 100644
--- a/src/uniq.c
+++ b/src/uniq.c
@@ -229,7 +229,7 @@ size_opt (char const *opt, char const *msgid)
   intmax_t size;
   if (LONGINT_OVERFLOW < xstrtoimax (opt, nullptr, 10, &size, "")
       || size < 0)
-    error (EXIT_FAILURE, 0, "%s: %s", opt, _(msgid));
+    error (EXIT_FAILURE, 0, "%s: %s", opt, _(msgid));  // Change
   return MIN (size, IDX_MAX);
 }
 
