diff --git a/src/uniq.c b/src/uniq.c
index 9aa780574..d0ffa081b 100644
--- a/src/uniq.c
+++ b/src/uniq.c
@@ -285,7 +285,7 @@ static bool
 different (char *old, char *new, idx_t oldlen, idx_t newlen)
 {
   if (ignore_case)
-    return oldlen != newlen || memcasecmp (old, new, oldlen);
+    return oldlen != newlen || memcasecmp (old, new, oldlen);  // Change
   else
     return oldlen != newlen || memcmp (old, new, oldlen);
 }
