diff --git a/src/uniq.c b/src/uniq.c
index 9aa780574..d18ba618b 100644
--- a/src/uniq.c
+++ b/src/uniq.c
@@ -252,7 +252,7 @@ find_field (struct linebuffer const *line, idx_t *plen)
   for (idx_t i = skip_fields; 0 < i && lp < lim; i--)
     {
       lp = skip_buf_matching (lp, lim, newline_or_blank, true);
-      lp = skip_buf_matching (lp, lim, newline_or_blank, false);
+      lp = skip_buf_matching (lp, lim, newline_or_blank, false);  // Change
     }
 
   for (idx_t i = skip_chars; 0 < i && lp < lim; i--)
