diff --git a/projects/PerformanceInteractionDetection/DegreeComplex/main.cpp b/projects/PerformanceInteractionDetection/DegreeComplex/main.cpp
index f446f22..dc9a4f4 100644
--- a/projects/PerformanceInteractionDetection/DegreeComplex/main.cpp
+++ b/projects/PerformanceInteractionDetection/DegreeComplex/main.cpp
@@ -35,6 +35,8 @@ int main(int argc, char *argv[]) {
       t *= 2;
   }
 
+  ___REGION_START __RT_High "hot_code"
   fp_util::sleep_for_millisecs(1000 + t * 100);
+  ___REGION_END __RT_High "hot_code"
   return 0;
 }
