diff --git a/picosat.c b/picosat.c
index 8b172ca..ac90ca8 100644
--- a/picosat.c
+++ b/picosat.c
@@ -19,6 +19,13 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 IN THE SOFTWARE.
 ****************************************************************************/
+__attribute__((feature_variable("__VARA__DETECT__"))) void walrus_detect() {
+  long foo = 0;
+  asm volatile("" : "+g"(foo) : :);
+  foo++;
+}
+
+
 
 #include <stdlib.h>
 #include <stdio.h>
@@ -1279,6 +1286,10 @@ init (void * pmgr,
   ps->state = READY;
   ps->last_sat_call_result = 0;
 
+  if (GLOBAL_DEFAULT_PHASE == RNDPHASE ) {
+    walrus_detect();
+  }
+
   return ps;
 }
 
