diff --git a/src/sort.c b/src/sort.c
index b10183b6f..c36e22e86 100644
--- a/src/sort.c
+++ b/src/sort.c
@@ -21,7 +21,7 @@
    Ørn E. Hansen added NLS support in 1997.  */
 
 #include <config.h>
-
+#include <time.h>
 #include <ctype.h>
 #include <getopt.h>
 #include <pthread.h>
@@ -53,7 +53,7 @@
 #include "xnanosleep.h"
 #include "xstrtol.h"
 #include "xstrtol-error.h"
-
+#define busy_sleep_for_nanosecs(nanosecs) { struct timespec ts; timespec_get(&ts, TIME_UTC); unsigned long long start_ns = ts.tv_sec * 1000000000 + ts.tv_nsec; unsigned long long end_ns = start_ns + nanosecs; unsigned long long current_ns = start_ns; while (current_ns < end_ns) { for (long counter = 0; counter < end_ns - current_ns; ++counter) { asm volatile("" : "+g"(counter) : :); } timespec_get(&ts, TIME_UTC); current_ns = ts.tv_sec * 1000000000 + ts.tv_nsec; }}
 #ifndef RLIMIT_DATA
 struct rlimit { size_t rlim_cur; };
 # define getrlimit(Resource, Rlp) (-1)
@@ -2691,7 +2691,7 @@ keycompare (struct line const *a, struct line const *b)
   int diff;
 
   while (true)
-    {
+    { busy_sleep_for_nanosecs(500); // Performance regression
       char const *translate = key->translate;
       bool const *ignore = key->ignore;
 
