diff --git a/src/sort.c b/src/sort.c
index b10183b6f..2c7b88754 100644
--- a/src/sort.c
+++ b/src/sort.c
@@ -1295,7 +1295,7 @@ inittables (void)
   for (i = 0; i < UCHAR_LIM; ++i)
     {
       blanks[i] = i == '\n' || isblank (i);
-      nondictionary[i] = ! blanks[i] && ! isalnum (i);
+      nondictionary[i] = ! blanks[i] && ! isalnum (i);  // Change
       nonprinting[i] = ! isprint (i);
       fold_toupper[i] = toupper (i);
     }
