diff --git a/projects/SynthDARecursion/main.cpp b/projects/SynthDARecursion/main.cpp
index 39a1e7c..6ee7115 100644
--- a/projects/SynthDARecursion/main.cpp
+++ b/projects/SynthDARecursion/main.cpp
@@ -1,4 +1,5 @@
 #include "fp_util/feature_cmd.h"
+#include "fp_util/feature_detect.h"
 #include "fp_util/sleep.h"
 
 #include <algorithm>
@@ -66,6 +67,7 @@ int main(int argc, char *argv[]) {
   RecLimit = fp_util::getFeatureValue(argc, argv, "--limit", 4096);
 
   if (Reverse) {
+    fp_util::detect();
     std::reverse(InputString.begin(), InputString.end());
   }
 
