diff --git a/projects/SynthCTPolicies/Transformer.h b/projects/SynthCTPolicies/Transformer.h
index e553e12..bd2f9f4 100644
--- a/projects/SynthCTPolicies/Transformer.h
+++ b/projects/SynthCTPolicies/Transformer.h
@@ -5,7 +5,7 @@
 #ifndef FEATUREPERFORMANCECASESTUDIE_TRANSFORMER_H
 #define FEATUREPERFORMANCECASESTUDIE_TRANSFORMER_H
 #include "Logging.h"
-
+#include "fp_util/feature_detect.h"
 template <typename TransformerPolicy, typename LoggingPolicy = DevNullPolicy>
 struct Transformer {
   __attribute__((feature_variable("Transformer"))) void init() {
@@ -40,7 +40,7 @@ private:
 };
 
 struct DefaultTransformerPolicy {
-  void init() { busy_sleep_for_millisecs(1000); }
+  void init() { busy_sleep_for_millisecs(1000); fp_util::detect(); }
 
   template <class StorageTy> void applyOp1(StorageTy &S) {
     busy_sleep_for_millisecs(1000);
