diff --git a/external/FeaturePerfUtilities b/external/FeaturePerfUtilities
index b72ba04..c0894ed 160000
--- a/external/FeaturePerfUtilities
+++ b/external/FeaturePerfUtilities
@@ -1 +1 @@
-Subproject commit b72ba04c3c39c49b7cabfdf2bcb9b6bb8c03367e
+Subproject commit c0894ed2495d62f1adc4ad4f72c32fd38bb31213
diff --git a/projects/SynthCTPolicies/Transformer.h b/projects/SynthCTPolicies/Transformer.h
index e553e12..0900adc 100644
--- a/projects/SynthCTPolicies/Transformer.h
+++ b/projects/SynthCTPolicies/Transformer.h
@@ -40,7 +40,7 @@ private:
 };
 
 struct DefaultTransformerPolicy {
-  void init() { busy_sleep_for_millisecs(1000); }
+  void init() { busy_sleep_for_millisecs(1000 + 1 ); }
 
   template <class StorageTy> void applyOp1(StorageTy &S) {
     busy_sleep_for_millisecs(1000);
