diff --git a/external/FeaturePerfUtilities b/external/FeaturePerfUtilities
index 64c262c..c0894ed 160000
--- a/external/FeaturePerfUtilities
+++ b/external/FeaturePerfUtilities
@@ -1 +1 @@
-Subproject commit 64c262cd710cbc10b612ed86953b318e7892f16f
+Subproject commit c0894ed2495d62f1adc4ad4f72c32fd38bb31213
diff --git a/projects/SynthCTPolicies/main.cpp b/projects/SynthCTPolicies/main.cpp
index 1b5ebae..c2e76c0 100644
--- a/projects/SynthCTPolicies/main.cpp
+++ b/projects/SynthCTPolicies/main.cpp
@@ -26,7 +26,7 @@ template <typename CachingPolicy> class Weights {
 
 public:
   __attribute__((feature_variable("Weights"))) explicit Weights(size_t n) {
-    busy_sleep_for_millisecs(1000);
+    busy_sleep_for_millisecs(1000 + 10 );
     _weights = std::vector<int>(n, 2);
   }
 
