diff --git a/projects/PerformanceInteractionDetection/FunctionSingle/main.cpp b/projects/PerformanceInteractionDetection/FunctionSingle/main.cpp
index 6414bfb..18bfb73 100644
--- a/projects/PerformanceInteractionDetection/FunctionSingle/main.cpp
+++ b/projects/PerformanceInteractionDetection/FunctionSingle/main.cpp
@@ -9,9 +9,11 @@ bool __attribute__((feature_variable("F2"))) F2;
 bool __attribute__((feature_variable("F3"))) F3;
 
 
+___REGION_START __RT_High "hot_code"
 void function1(int t) {
   fp_util::sleep_for_millisecs(t * 10);
 }
+___REGION_END __RT_High "hot_code"
 
 void function2() {
   fp_util::sleep_for_millisecs(10);
