diff --git a/projects/PerformanceInteractionDetection/DegreeLow/main.cpp b/projects/PerformanceInteractionDetection/DegreeLow/main.cpp
index abde64c..65240f0 100644
--- a/projects/PerformanceInteractionDetection/DegreeLow/main.cpp
+++ b/projects/PerformanceInteractionDetection/DegreeLow/main.cpp
@@ -34,6 +34,8 @@ int main(int argc, char *argv[]) {
     t *= 2;
   }
 
+  ___REGION_START __RT_High "hot_code"
   fp_util::sleep_for_millisecs(100 + t);
+  ___REGION_END __RT_High "hot_code"
   return 0;
 }
