diff --git a/projects/PerformanceInteractionDetection/DegreeHigh/main.cpp b/projects/PerformanceInteractionDetection/DegreeHigh/main.cpp
index ba87834..8ae344c 100644
--- a/projects/PerformanceInteractionDetection/DegreeHigh/main.cpp
+++ b/projects/PerformanceInteractionDetection/DegreeHigh/main.cpp
@@ -28,7 +28,7 @@ int main(int argc, char *argv[]) {
   F9 = fp_util::isFeatureEnabled(argc, argv, "--f9");
   F10 = fp_util::isFeatureEnabled(argc, argv, "--f10");
 
-  int t = 10;
+  int t = 15;
 
   if (F1) {
     if (F2) {
