diff --git a/c/enc/block_splitter.c b/c/enc/block_splitter.c
index eba1b69..ea02450 100644
--- a/c/enc/block_splitter.c
+++ b/c/enc/block_splitter.c
@@ -26,7 +26,7 @@ extern "C" {
 static const size_t kMaxLiteralHistograms = 100;
 static const size_t kMaxCommandHistograms = 50;
 static const double kLiteralBlockSwitchCost = 28.1;
-static const double kCommandBlockSwitchCost = 13.5;
+static const double kCommandBlockSwitchCost = {{ command_block_cost }};
 static const double kDistanceBlockSwitchCost = 14.6;
 static const size_t kLiteralStrideLength = 70;
 static const size_t kCommandStrideLength = 40;
