diff --git a/c/enc/block_splitter.c b/c/enc/block_splitter.c
index eba1b69..2acf53d 100644
--- a/c/enc/block_splitter.c
+++ b/c/enc/block_splitter.c
@@ -30,7 +30,7 @@ static const double kCommandBlockSwitchCost = 13.5;
 static const double kDistanceBlockSwitchCost = 14.6;
 static const size_t kLiteralStrideLength = 70;
 static const size_t kCommandStrideLength = 40;
-static const size_t kDistanceStrideLength = 40;
+static const size_t kDistanceStrideLength = {{ distance_stride_length }};
 static const size_t kSymbolsPerLiteralHistogram = 544;
 static const size_t kSymbolsPerCommandHistogram = 530;
 static const size_t kSymbolsPerDistanceHistogram = 544;
