diff --git a/c/enc/encode.c b/c/enc/encode.c
index 96c517b..3ee6784 100644
--- a/c/enc/encode.c
+++ b/c/enc/encode.c
@@ -437,7 +437,7 @@ static BROTLI_BOOL ShouldCompress(
     if ((double)num_literals > 0.99 * (double)bytes) {
       uint32_t literal_histo[256] = { 0 };
       static const uint32_t kSampleRate = 13;
-      static const double kMinEntropy = 7.92;
+      static const double kMinEntropy = {{ min_entropy }};
       const double bit_cost_threshold =
           (double)bytes * kMinEntropy / kSampleRate;
       size_t t = (bytes + kSampleRate - 1) / kSampleRate;
