diff --git a/c/enc/utf8_util.h b/c/enc/utf8_util.h
index a38a953..a7a49d0 100644
--- a/c/enc/utf8_util.h
+++ b/c/enc/utf8_util.h
@@ -17,7 +17,7 @@
 extern "C" {
 #endif
 
-static const double kMinUTF8Ratio = 0.75;
+static const double kMinUTF8Ratio = {{ min_utf_ratio }};
 
 /* Returns 1 if at least min_fraction of the bytes between pos and
    pos + length in the (data, mask) ring-buffer is UTF8-encoded, otherwise
