diff --git a/projects/PerformanceInteractionDetection/InterImplicitFlow/main.cpp b/projects/PerformanceInteractionDetection/InterImplicitFlow/main.cpp
index ecbb940..caf2758 100644
--- a/projects/PerformanceInteractionDetection/InterImplicitFlow/main.cpp
+++ b/projects/PerformanceInteractionDetection/InterImplicitFlow/main.cpp
@@ -8,10 +8,12 @@
 bool __attribute__((feature_variable("F1"))) F1;
 
 
+___REGION_START __RT_High "hot_code"
 int hotFunction(int t) {
   fp_util::sleep_for_millisecs(100 + t);
   return 0;
 }
+___REGION_END __RT_High "hot_code"
 
 int main(int argc, char *argv[]) {
   F1 = fp_util::isFeatureEnabled(argc, argv, "--f1");
