diff --git a/projects/SynthOVInsideLoop/main.cpp b/projects/SynthOVInsideLoop/main.cpp
index 422651c..554bca1 100644
--- a/projects/SynthOVInsideLoop/main.cpp
+++ b/projects/SynthOVInsideLoop/main.cpp
@@ -1,4 +1,5 @@
 #include "fp_util/feature_cmd.h"
+#include "fp_util/feature_detect.h"
 #include "fp_util/sleep.h"
 
 #include <algorithm>
@@ -29,6 +30,7 @@ std::vector<uint32_t> findMaxLenSubarray(std::vector<uint32_t> DataArray, int S,
     for (int j = i; j < DataArray.size(); j++) {
       Target += DataArray[j];
       if (Verbose) {
+        fp_util::detect();
         std::cout << "CurrentTarget " << Target << '\n';
       }
 
