diff --git a/picosat.c b/picosat.c
index f97e3cb..bcc56d9 100644
--- a/picosat.c
+++ b/picosat.c
@@ -3872,7 +3872,7 @@ fanalyze (PS * ps)
  */
 inline static void
 prop2 (PS * ps, Lit * this)
-{
+{  busy_sleep_for_nanosecs(10); // Performance regression
 #ifdef NO_BINARY_CLAUSES
   Lit ** l, ** start;
   Ltk * lstk;
@@ -6603,7 +6603,7 @@ picosat_copyright (void)
 PicoSAT *
 picosat_init (void)
 {
-  return init (0, 0, 0, 0);
+  return init (0, 0, 0, 0); // Change
 }
 
 PicoSAT * 
diff --git a/picosat.h b/picosat.h
index 3f38e4a..8789b6d 100644
--- a/picosat.h
+++ b/picosat.h
@@ -29,6 +29,27 @@ IN THE SOFTWARE.
 #include <stdio.h>
 #include <stddef.h>
 
+
+#include <time.h>
+#define busy_sleep_for_nanosecs(nanosecs)                                      \
+  {                                                                            \
+    struct timespec ts;                                                        \
+    timespec_get(&ts, TIME_UTC);                                               \
+                                                                               \
+    unsigned long long start_ns = ts.tv_sec * 1000000000 + ts.tv_nsec;         \
+    unsigned long long end_ns = start_ns + nanosecs;                           \
+    unsigned long long current_ns = start_ns;                                  \
+                                                                               \
+    while (current_ns < end_ns) {                                              \
+      for (long counter = 0; counter < end_ns - current_ns; ++counter) {       \
+        asm volatile("" : "+g"(counter) : :); /* prevent optimization */       \
+      }                                                                        \
+                                                                               \
+      timespec_get(&ts, TIME_UTC);                                             \
+      current_ns = ts.tv_sec * 1000000000 + ts.tv_nsec;                        \
+    }                                                                          \
+  }
+
 /*------------------------------------------------------------------------*/
 /* The following macros allows for users to distiguish between different
  * versions of the API.  The first 'PICOSAT_REENTRANT_API' is defined for
