diff --git a/projects/SynthSAContextSensitivity/main.cpp b/projects/SynthSAContextSensitivity/main.cpp
index 7e9d37b..932504a 100644
--- a/projects/SynthSAContextSensitivity/main.cpp
+++ b/projects/SynthSAContextSensitivity/main.cpp
@@ -72,6 +72,7 @@ int main(int argc, char *argv[]) {
 
   if (boundsChecks(MemoryLimit) < 256) { // 0 =< x < 256
     if (Compress) {
+      fp_util::sleep_for_secs(13);
       Result = compute(UserInput, 9); // leads to less imprecise results
     } else {
       Result = compute(UserInput, 8); // leads to imprecise results
