diff --git a/projects/SynthSAContextSensitivity/main.cpp b/projects/SynthSAContextSensitivity/main.cpp
index 344a2e2..6f602f1 100644
--- a/projects/SynthSAContextSensitivity/main.cpp
+++ b/projects/SynthSAContextSensitivity/main.cpp
@@ -1,4 +1,5 @@
 #include "fp_util/feature_cmd.h"
+#include "fp_util/feature_detect.h"
 #include "fp_util/sleep.h"
 
 #include <bit>
@@ -77,6 +78,7 @@ int main(int argc, char *argv[]) {
       Result = compute(UserInput, 8); // leads to imprecise results
     }
   } else if (boundsChecks(MemoryLimit) < 1024) { // 256 =< x < 1024
+    fp_util::detect();
     Result = compute(UserInput, 16);             // leads to imprecise results
   } else {
     Result = compute(UserInput, 250);
