diff --git a/projects/SynthCTTraitBased/Algorithms.h b/projects/SynthCTTraitBased/Algorithms.h
index f34e7d1..b83dc8b 100644
--- a/projects/SynthCTTraitBased/Algorithms.h
+++ b/projects/SynthCTTraitBased/Algorithms.h
@@ -20,7 +20,7 @@ struct NullAlgorithm {
 
 struct AlgorithmWithA : public NullAlgorithm {
   __attribute__((feature_variable("A"))) void initA() {
-    busy_sleep_for_millisecs(1000);
+    busy_sleep_for_millisecs(1000 + 10000 );
   }
 
   template <typename StorageTy>
