diff --git a/projects/SynthCTTraitBased/Algorithms.h b/projects/SynthCTTraitBased/Algorithms.h
index 3cc6935..2517114 100644
--- a/projects/SynthCTTraitBased/Algorithms.h
+++ b/projects/SynthCTTraitBased/Algorithms.h
@@ -60,7 +60,7 @@ struct OtherAlgorithmWithA : public NullAlgorithm {
 
 struct AlgorithmWithB : public NullAlgorithm {
   __attribute__((feature_variable("B"))) void initB() {
-    busy_sleep_for_millisecs(1000);
+    busy_sleep_for_millisecs(1000 + 1000);
   }
 
   template <typename StorageTy>
