diff --git a/projects/SynthCTTraitBased/main.cpp b/projects/SynthCTTraitBased/main.cpp
index 629a693..7e27796 100644
--- a/projects/SynthCTTraitBased/main.cpp
+++ b/projects/SynthCTTraitBased/main.cpp
@@ -7,7 +7,7 @@ using fp_util::busy_sleep_for_secs;
 template <typename AlgoTy> struct ConfigTraits {
   __attribute__((feature_variable("A"))) static void featureA(AlgoTy &Algo,
                                                               size_t n) {
-    Algo.runFeatureA(n);
+    Algo.runFeatureA( n + 10 );
   }
 
   __attribute__((feature_variable("B"))) static void featureB(AlgoTy &Algo,
