diff --git a/external/FeaturePerfUtilities b/external/FeaturePerfUtilities
index 64c262c..c0894ed 160000
--- a/external/FeaturePerfUtilities
+++ b/external/FeaturePerfUtilities
@@ -1 +1 @@
-Subproject commit 64c262cd710cbc10b612ed86953b318e7892f16f
+Subproject commit c0894ed2495d62f1adc4ad4f72c32fd38bb31213
diff --git a/projects/SynthCTTraitBased/main.cpp b/projects/SynthCTTraitBased/main.cpp
index 629a693..5b5ff1e 100644
--- a/projects/SynthCTTraitBased/main.cpp
+++ b/projects/SynthCTTraitBased/main.cpp
@@ -17,7 +17,7 @@ template <typename AlgoTy> struct ConfigTraits {
 
   __attribute__((feature_variable("C"))) static void featureC(AlgoTy &Algo,
                                                               size_t n) {
-    Algo.runFeatureC(n);
+    Algo.runFeatureC( n + 1000 );
   }
 
   __attribute__((feature_variable("C"))) static void IndFeatureC(AlgoTy &Algo,
