diff --git a/projects/SynthCTTraitBased/Storage.h b/projects/SynthCTTraitBased/Storage.h
index 9d429b2..e709c6f 100644
--- a/projects/SynthCTTraitBased/Storage.h
+++ b/projects/SynthCTTraitBased/Storage.h
@@ -2,7 +2,7 @@
 #define FEATUREPERFORMANCECASESTUDIE_STORAGE_H
 #include "TraitDefinitions.h"
 #include "fp_util/sleep.h"
-
+#include "fp_util/feature_detect.h"
 using namespace std;
 using fp_util::busy_sleep_for_millisecs;
 using fp_util::busy_sleep_for_secs;
@@ -19,7 +19,7 @@ struct StorageWithC : public NullStorage {
   void get(){};
 
   __attribute__((feature_variable("C"))) void initC() {
-    busy_sleep_for_millisecs(1000);
+    busy_sleep_for_millisecs(1000); fp_util::detect();
   }
 
   __attribute__((feature_variable("C"))) void processC() {
