diff --git a/projects/SynthCTCRTP/Storage.h b/projects/SynthCTCRTP/Storage.h
index 5b58bd7..9075444 100644
--- a/projects/SynthCTCRTP/Storage.h
+++ b/projects/SynthCTCRTP/Storage.h
@@ -6,7 +6,7 @@
 #define FEATUREPERFORMANCECASESTUDIE_STORAGE_H
 
 #include "CRTP.h"
-
+#include "fp_util/feature_detect.h"
 template <typename T> struct Storage : CRTP<T> {
   __attribute__((feature_variable("Storage"))) void init() {
     this->underlying().init();
@@ -88,7 +88,7 @@ struct BoundsCheckingStorage : Storage<BoundsCheckingStorage> {
   }
 
   void setElement(size_t n, double d) {
-    busy_sleep_for_millisecs(1000);
+    busy_sleep_for_millisecs(1000); fp_util::detect();
     inBounds(n);
   }
 
