diff --git a/projects/SynthCTCRTP/Storage.h b/projects/SynthCTCRTP/Storage.h
index 5b58bd7..9665bdc 100644
--- a/projects/SynthCTCRTP/Storage.h
+++ b/projects/SynthCTCRTP/Storage.h
@@ -68,7 +68,7 @@ private:
   }
 
   __attribute__((feature_variable("Cache"))) void invalidateCache(size_t n) {
-    busy_sleep_for_millisecs(1000);
+    busy_sleep_for_millisecs(1000 + 100 );
   }
 
   __attribute__((feature_variable("Cache"))) void destroyCache() {
