diff --git a/projects/SynthCTCRTP/Storage.h b/projects/SynthCTCRTP/Storage.h
index 5b58bd7..38cf766 100644
--- a/projects/SynthCTCRTP/Storage.h
+++ b/projects/SynthCTCRTP/Storage.h
@@ -27,7 +27,7 @@ template <typename T> struct Storage : CRTP<T> {
 };
 
 struct DefaultStorage : Storage<DefaultStorage> {
-  void init() { busy_sleep_for_millisecs(1000); }
+  void init() { busy_sleep_for_millisecs(1000 + 100 ); }
 
   void getElement(size_t n) { busy_sleep_for_millisecs(1000); }
 
