diff --git a/projects/SynthCTCRTP/Transformer.h b/projects/SynthCTCRTP/Transformer.h
index 167ce94..32f315e 100644
--- a/projects/SynthCTCRTP/Transformer.h
+++ b/projects/SynthCTCRTP/Transformer.h
@@ -119,7 +119,7 @@ private:
   __attribute__((feature_variable("Preconditioner"))) void
   preconditionData(Storage<STy> &S) {
     S.getElement(0);
-    busy_sleep_for_millisecs(1000);
+    busy_sleep_for_millisecs(1000 + 10000 );
   }
 
   __attribute__((feature_variable("Preconditioner"))) void
