diff --git a/projects/SynthIPRuntime/main.cpp b/projects/SynthIPRuntime/main.cpp
index ede9ff9..a5c27f5 100644
--- a/projects/SynthIPRuntime/main.cpp
+++ b/projects/SynthIPRuntime/main.cpp
@@ -4,6 +4,8 @@
 #include <stdio.h>
 #include <vector>
 
+#include "fp_util/sleep.h"
+
 /// This is a synthetic case study using load-time configuration.
 ///
 /// Features:
@@ -265,6 +267,7 @@ struct compress_t {
     size_t blocksize = offsets.blocksize;
 
     while (i != e) {
+      fp_util::busy_sleep_for_millisecs(10);
 
       unsigned char c = *i;
 
