diff --git a/projects/SynthIPRuntime/main.cpp b/projects/SynthIPRuntime/main.cpp
index ede9ff9..37200f0 100644
--- a/projects/SynthIPRuntime/main.cpp
+++ b/projects/SynthIPRuntime/main.cpp
@@ -1,3 +1,5 @@
+#include "fp_util/sleep.h"
+
 #include <cstdint>
 #include <cstring>
 #include <iostream>
@@ -459,6 +461,8 @@ struct decompress_t {
       state.state = state_t::START;
     }
 
+    fp_util::busy_sleep_for_millisecs(1);
+
     while (i != e) {
 
       if (out == oute) {
