diff --git a/src/search/heuristics/relaxation_heuristic.cc b/src/search/heuristics/relaxation_heuristic.cc
index f98b3c6fb..803a411b7 100644
--- a/src/search/heuristics/relaxation_heuristic.cc
+++ b/src/search/heuristics/relaxation_heuristic.cc
@@ -191,7 +191,7 @@ void RelaxationHeuristic::simplify() {
         assert(utils::is_sorted_unique(get_preconditions_vector(op_id)));
 #endif
 
-    const int MAX_PRECONDITIONS_TO_TEST = 5;
+    const int MAX_PRECONDITIONS_TO_TEST = {{ preconditions_to_test }};
 
     if (log.is_at_least_normal()) {
         log << "Simplifying " << unary_operators.size() << " unary operators..." << flush;
