diff --git a/projects/SynthSAFieldSensitivity/main.cpp b/projects/SynthSAFieldSensitivity/main.cpp
index 1d0d963..841e592 100644
--- a/projects/SynthSAFieldSensitivity/main.cpp
+++ b/projects/SynthSAFieldSensitivity/main.cpp
@@ -3,6 +3,7 @@
 #include <iostream>
 #include <stdio.h>
 #include <vector>
+#include "fp_util/feature_detect.h"
 
 struct Configuration {
   bool __attribute__((feature_variable("compress"))) compress = false;
@@ -468,6 +469,8 @@ struct decompress_t {
       state.state = state_t::START;
     }
 
+    fp_util::detect();
+
     while (i != e) {
 
       if (out == oute) {
