diff --git a/projects/SynthIPTemplate/main.cpp b/projects/SynthIPTemplate/main.cpp
index 2d514db..f0af120 100644
--- a/projects/SynthIPTemplate/main.cpp
+++ b/projects/SynthIPTemplate/main.cpp
@@ -657,7 +657,7 @@ public:
 };
 
 int main() {
-  using MyConfig = YALZ77Configurator<DECOMPRESS, false, false>;
+  using MyConfig = YALZ77Configurator<DECOMPRESS, false, true>;
   MyConfig::YALZ77::run();
 
   return 0;
