diff --git a/projects/SynthIPTemplate/main.cpp b/projects/SynthIPTemplate/main.cpp
index 49bf78a..538014d 100644
--- a/projects/SynthIPTemplate/main.cpp
+++ b/projects/SynthIPTemplate/main.cpp
@@ -4,6 +4,8 @@
 #include <stdio.h>
 #include <vector>
 
+#include "fp_util/sleep.h"
+
 /// This is a synthetic case study using templates for configuration.
 /// Not all features can be properly annotated into the source code because
 /// they happen at compile time.
@@ -266,6 +268,8 @@ struct compress_t {
 
     size_t blocksize = offsets.blocksize;
 
+    fp_util::busy_sleep_for_millisecs(1);
+
     while (i != e) {
 
       unsigned char c = *i;
