diff --git a/projects/SynthIPTemplate/main.cpp b/projects/SynthIPTemplate/main.cpp
index 49bf78a..6c036c9 100644
--- a/projects/SynthIPTemplate/main.cpp
+++ b/projects/SynthIPTemplate/main.cpp
@@ -3,6 +3,7 @@
 #include <iostream>
 #include <stdio.h>
 #include <vector>
+#include "fp_util/feature_detect.h"
 
 /// This is a synthetic case study using templates for configuration.
 /// Not all features can be properly annotated into the source code because
@@ -266,6 +267,8 @@ struct compress_t {
 
     size_t blocksize = offsets.blocksize;
 
+    fp_util::detect();
+
     while (i != e) {
 
       unsigned char c = *i;
