diff --git a/projects/SynthIPTemplate2/main.cpp b/projects/SynthIPTemplate2/main.cpp
index 97d463e..5a63b68 100644
--- a/projects/SynthIPTemplate2/main.cpp
+++ b/projects/SynthIPTemplate2/main.cpp
@@ -675,7 +675,7 @@ struct YALZ77Configurator {
 };
 
 int main() {
-  using MyConfig = YALZ77Configurator<DECOMPRESS, false, false>;
+  using MyConfig = YALZ77Configurator<COMPRESS, false, true>;
   MyConfig::YALZ77::run(getSearchlen(MyConfig::fast),
                         getBlocksize(MyConfig::small),
                         getBufsize(MyConfig::small, MyConfig::mode));
