diff --git a/projects/SynthIPTemplate2/main.cpp b/projects/SynthIPTemplate2/main.cpp
index 9e1283e..be68be8 100644
--- a/projects/SynthIPTemplate2/main.cpp
+++ b/projects/SynthIPTemplate2/main.cpp
@@ -4,6 +4,8 @@
 #include <stdio.h>
 #include <vector>
 
+#include "fp_util/sleep.h"
+
 /// This is a synthetic case study using templates for configuration.
 /// This implementation defers some configuration-dependent computations
 /// until runtime. Therefore, all features can be annotated.
@@ -266,6 +268,8 @@ struct compress_t {
 
     size_t blocksize = offsets.blocksize;
 
+    fp_util::busy_sleep_for_millisecs(1);
+
     while (i != e) {
 
       unsigned char c = *i;
