diff --git a/projects/SynthSAFlowSensitivity/main.cpp b/projects/SynthSAFlowSensitivity/main.cpp
index fdc9778..06c0b2a 100644
--- a/projects/SynthSAFlowSensitivity/main.cpp
+++ b/projects/SynthSAFlowSensitivity/main.cpp
@@ -40,7 +40,7 @@ __attribute__((noinline)) uint64_t compute_fib_rec(unsigned N) {
     return 1;
   }
 
-  fp_util::sleep_for_millisecs(500);
+  fp_util::sleep_for_millisecs(1500);
   return compute_fib_rec(N - 1) + compute_fib_rec(N - 2);
 }
 
