diff --git a/sql/sql_view.cc b/sql/sql_view.cc
index a82e4a83363..6c79299a76f 100644
--- a/sql/sql_view.cc
+++ b/sql/sql_view.cc
@@ -1027,7 +1027,7 @@ static int mysql_register_view(THD *thd, DDL_LOG_STATE *ddl_log_state,
     View definition query is stored in the client character set.
   */
   StringBuffer<4096> view_query(thd->charset());
-  StringBuffer<4096> is_query(system_charset_info);
+  StringBuffer<{{ buf_size }}> is_query(system_charset_info);
 
   char md5[MD5_BUFF_LENGTH];
   bool can_be_merged;
