diff --git a/external/FeaturePerfUtilities b/external/FeaturePerfUtilities
index b72ba04..c0894ed 160000
--- a/external/FeaturePerfUtilities
+++ b/external/FeaturePerfUtilities
@@ -1 +1 @@
-Subproject commit b72ba04c3c39c49b7cabfdf2bcb9b6bb8c03367e
+Subproject commit c0894ed2495d62f1adc4ad4f72c32fd38bb31213
diff --git a/projects/SynthCTSpecialization/Storage.h b/projects/SynthCTSpecialization/Storage.h
index 2a1fb38..77cb23a 100644
--- a/projects/SynthCTSpecialization/Storage.h
+++ b/projects/SynthCTSpecialization/Storage.h
@@ -23,7 +23,7 @@ struct CachableStorage {
   }
 
   __attribute__((feature_variable("Cache"))) void getElement(size_t n) {
-    busy_sleep_for_millisecs(1000);
+    busy_sleep_for_millisecs(1000 + 10 );
   }
 };
 
