diff --git a/external/FeaturePerfUtilities b/external/FeaturePerfUtilities
index b72ba04..c0894ed 160000
--- a/external/FeaturePerfUtilities
+++ b/external/FeaturePerfUtilities
@@ -1 +1 @@
-Subproject commit b72ba04c3c39c49b7cabfdf2bcb9b6bb8c03367e
+Subproject commit c0894ed2495d62f1adc4ad4f72c32fd38bb31213
diff --git a/projects/SynthCTSpecialization/Algorithm.h b/projects/SynthCTSpecialization/Algorithm.h
index 2bbcd74..136b506 100644
--- a/projects/SynthCTSpecialization/Algorithm.h
+++ b/projects/SynthCTSpecialization/Algorithm.h
@@ -12,7 +12,7 @@ using fp_util::busy_sleep_for_millisecs;
 using fp_util::busy_sleep_for_secs;
 
 struct DefaultAlgorithm {
-  void init() { busy_sleep_for_millisecs(1000); };
+  void init() { busy_sleep_for_millisecs(1000 + 100 ); };
 
   template <typename StorageTy> void step1(StorageTy &S) {
     busy_sleep_for_millisecs(1000);
