diff --git a/projects/SynthDAVirtualInheritance/main.cpp b/projects/SynthDAVirtualInheritance/main.cpp
index 90d60c3..bd05566 100644
--- a/projects/SynthDAVirtualInheritance/main.cpp
+++ b/projects/SynthDAVirtualInheritance/main.cpp
@@ -71,7 +71,7 @@ public:
     int High = LocalData.size() - 1;
 
     while (Low < High) {
-      fp_util::sleep_for_millisecs(100);
+      fp_util::sleep_for_millisecs(200);
       if (LocalData[Low] + LocalData[High] == TargetSum) {
         return {
             std::tuple<uint32_t, uint32_t>{LocalData[Low], LocalData[High]}};
