diff --git a/external/FeaturePerfUtilities b/external/FeaturePerfUtilities
index b72ba04..f570808 160000
--- a/external/FeaturePerfUtilities
+++ b/external/FeaturePerfUtilities
@@ -1 +1 @@
-Subproject commit b72ba04c3c39c49b7cabfdf2bcb9b6bb8c03367e
+Subproject commit f570808f511b511fd389db165d85a3d02c06d639
diff --git a/projects/SynthDADynamicDispatch/main.cpp b/projects/SynthDADynamicDispatch/main.cpp
index 90d60c3..89ca905 100644
--- a/projects/SynthDADynamicDispatch/main.cpp
+++ b/projects/SynthDADynamicDispatch/main.cpp
@@ -10,6 +10,8 @@
 #include <unordered_map>
 #include <vector>
 
+#include "fp_util/feature_detect.h"
+
 /// This synthetic case study targets virtual inheritance.
 ///
 /// --sorting  : use a sort-based searcher instead
@@ -43,6 +45,7 @@ public:
 
     for (int i = 0; i < Data.size() - 1; i++) {
       for (int j = i + 1; j < Data.size(); j++) {
+        fp_util::detect();
         fp_util::sleep_for_millisecs(100);
         if (Data[i] + Data[j] == TargetSum) {
           return {std::tuple<uint32_t, uint32_t>{Data[i], Data[j]}};
