diff --git a/storage/innobase/log/log0write.cc b/storage/innobase/log/log0write.cc
index f4d5b7fe87d..aa3ebfdab8c 100644
--- a/storage/innobase/log/log0write.cc
+++ b/storage/innobase/log/log0write.cc
@@ -1994,11 +1994,11 @@ static lsn_t log_writer_wait_on_checkpoint(log_t &log, lsn_t last_write_lsn,
 }
 
 static void log_writer_wait_on_archiver(log_t &log, lsn_t next_write_lsn) {
-  const int32_t SLEEP_BETWEEN_RETRIES_IN_US = 100; /* 100us */
+  const int32_t SLEEP_BETWEEN_RETRIES_IN_US = {{ sleep_retries }}; /* 100us */
 
-  const int32_t TIME_BETWEEN_WARNINGS_IN_US = 100000; /* 100ms */
+  const int32_t TIME_BETWEEN_WARNINGS_IN_US = {{ sleep_retries * 1000 }}; /* 100ms */
 
-  const int32_t TIME_UNTIL_ERROR_IN_US = 1000000; /* 1s */
+  const int32_t TIME_UNTIL_ERROR_IN_US = {{ sleep_retries * 10000 }}; /* 1s */
 
   ut_ad(log_writer_mutex_own(log));
 
