diff --git a/CMakeLists.txt b/CMakeLists.txt
index 03462271..fded5831 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1588,7 +1588,7 @@ if(BUILD_SHARED)
   if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug") # Why?
     option(WITH_PERF_TOOL "Build with perf-tools" ON)
   else()
-    option(WITH_PERF_TOOL "Build with perf-tools" OFF)
+    option(WITH_PERF_TOOL "Build with perf-tools" ON)
   endif()
 
   if(WITH_PERF_TOOL)
diff --git a/src/config.hpp b/src/config.hpp
index 2feb0d4e..a2ab8621 100644
--- a/src/config.hpp
+++ b/src/config.hpp
@@ -12,7 +12,7 @@ enum
     //  Number of new messages in message pipe needed to trigger new memory
     //  allocation. Setting this parameter to 256 decreases the impact of
     //  memory allocation by approximately 99.6%
-    message_pipe_granularity = 256,
+    message_pipe_granularity = {{ message_pipe_granularity }},
 
     //  Commands in pipe per allocation event.
     command_pipe_granularity = 16,
