diff --git a/projects/SynthIPCombined/main.cpp b/projects/SynthIPCombined/main.cpp
index b8a13df..33417d7 100644
--- a/projects/SynthIPCombined/main.cpp
+++ b/projects/SynthIPCombined/main.cpp
@@ -616,7 +616,7 @@ int main(int argc, char **argv) {
       decompress = true;
   }
 
-  using MyConfig = YALZ77Configurator<false, false>;
+  using MyConfig = YALZ77Configurator<true, true>;
 
   const size_t BUFSIZE = getBufsize(MyConfig::small, decompress);
 
