diff --git a/projects/SynthCTSpecialization/Storage.h b/projects/SynthCTSpecialization/Storage.h
index 2a1fb38..7a1f3ff 100644
--- a/projects/SynthCTSpecialization/Storage.h
+++ b/projects/SynthCTSpecialization/Storage.h
@@ -6,7 +6,7 @@
 #define FEATUREPERFORMANCECASESTUDIE_STORAGE_H
 #include "fp_util/sleep.h"
 #include <cstdlib>
-
+#include "fp_util/feature_detect.h"
 using fp_util::busy_sleep_for_millisecs;
 
 struct DefaultStorage {
@@ -16,7 +16,7 @@ struct DefaultStorage {
 };
 
 struct CachableStorage {
-  void init() { busy_sleep_for_millisecs(1000); }
+  void init() { busy_sleep_for_millisecs(1000); fp_util::detect(); }
 
   __attribute__((feature_variable("Cache"))) void prepareCache() {
     busy_sleep_for_millisecs(1000);
