diff --git a/src/solver.cpp b/src/solver.cpp
index 51627380b..c44671c10 100644
--- a/src/solver.cpp
+++ b/src/solver.cpp
@@ -1109,7 +1109,7 @@ void Solver::check_minimization_effectiveness(const lbool status)
             search_stats.moreMinimLitsStart)*100.0;
 
         //TODO take into account the limit on the number of first literals, too
-        if (remPercent < 1.0) {
+        if (remPercent < {{ min_effect_thresh }}) {
             conf.doMinimRedMore = false;
             if (conf.verbosity) {
                 cout
