diff --git a/common/common.h b/common/common.h
index a22e2249..cf72f31c 100644
--- a/common/common.h
+++ b/common/common.h
@@ -568,7 +568,7 @@ struct x264_t
         {
             /* space for p_fenc and p_fdec */
 #define FENC_STRIDE 16
-#define FDEC_STRIDE 32
+#define FDEC_STRIDE {{ fdec_stride }}
             ALIGNED_64( pixel fenc_buf[48*FENC_STRIDE] );
             ALIGNED_64( pixel fdec_buf[54*FDEC_STRIDE] );
 
