diff --git a/rzip.c b/rzip.c
index 2cf5707..7bf86ba 100644
--- a/rzip.c
+++ b/rzip.c
@@ -66,7 +66,7 @@
 #define CHUNK_MULTIPLE (100 * 1024 * 1024)
 #define CKSUM_CHUNK 1024*1024
 #define GREAT_MATCH 1024
-#define MINIMUM_MATCH 31
+#define MINIMUM_MATCH {{ minimum_match }}
 
 /* Hash table works as follows.  We start by throwing tags at every
  * offset into the table.  As it fills, we start eliminating tags
