diff --git a/projects/PerformanceInteractionDetection/FunctionAccumulating/main.cpp b/projects/PerformanceInteractionDetection/FunctionAccumulating/main.cpp
index 9a0e036..a6aeb76 100644
--- a/projects/PerformanceInteractionDetection/FunctionAccumulating/main.cpp
+++ b/projects/PerformanceInteractionDetection/FunctionAccumulating/main.cpp
@@ -9,17 +9,23 @@ bool __attribute__((feature_variable("F2"))) F2;
 bool __attribute__((feature_variable("F3"))) F3;
 
 
+___REGION_START __RT_High "hot_code"
 void function1(int r) {
   fp_util::sleep_for_millisecs(100 + r);
 }
+___REGION_END __RT_High "hot_code"
 
+___REGION_START __RT_High "hot_code"
 void function2(int s) {
   fp_util::sleep_for_millisecs(100 + s);
 }
+___REGION_END __RT_High "hot_code"
 
+___REGION_START __RT_High "hot_code"
 void function3(int t) {
   fp_util::sleep_for_millisecs(100 + t);
 }
+___REGION_END __RT_High "hot_code"
 
 int main(int argc, char *argv[]) {
   F1 = fp_util::isFeatureEnabled(argc, argv, "--f1");
