diff --git a/projects/PerformanceInteractionDetection/FunctionAccumulating/main.cpp b/projects/PerformanceInteractionDetection/FunctionAccumulating/main.cpp
index 9a0e036..91fb7e5 100644
--- a/projects/PerformanceInteractionDetection/FunctionAccumulating/main.cpp
+++ b/projects/PerformanceInteractionDetection/FunctionAccumulating/main.cpp
@@ -26,7 +26,7 @@ int main(int argc, char *argv[]) {
   F2 = fp_util::isFeatureEnabled(argc, argv, "--f2");
   F3 = fp_util::isFeatureEnabled(argc, argv, "--f3");
 
-  int r = 10;
+  int r = 15;
   int s = 10;
   int t = 10;
 
