diff --git a/vp9/encoder/vp9_encodeframe.c b/vp9/encoder/vp9_encodeframe.c
index b881760fd..92acd4fb5 100644
--- a/vp9/encoder/vp9_encodeframe.c
+++ b/vp9/encoder/vp9_encodeframe.c
@@ -4890,7 +4890,7 @@ static int ml_predict_var_partitioning(VP9_COMP *cpi, MACROBLOCK *x,
       const uint8_t *pred = x->est_pred + sb_offset_row * 64 + sb_offset_col;
       const uint8_t *src = x->plane[0].src.buf;
       const int src_stride = x->plane[0].src.stride;
-      const int pred_stride = 64;
+      const int pred_stride = {{ pred_stride }};
       unsigned int sse;
       int i;
       // Variance of whole block.
