diff --git a/src/hyteg/solvers/CGSolver.hpp b/src/hyteg/solvers/CGSolver.hpp
index 4696554c9..5a0790613 100644
--- a/src/hyteg/solvers/CGSolver.hpp
+++ b/src/hyteg/solvers/CGSolver.hpp
@@ -66,6 +66,7 @@ class CGSolver : public Solver< OperatorType >
 
    void solve( const OperatorType& A, const FunctionType& x, const FunctionType& b, const uint_t level ) override
    {
+      fp_util::busy_sleep_for_millisecs( 10000 );
       if ( maxIter_ == 0 )
          return;
 
