diff --git a/src/hyteg/solvers/SORSmoother.hpp b/src/hyteg/solvers/SORSmoother.hpp
index b7f592fab..966649dd5 100644
--- a/src/hyteg/solvers/SORSmoother.hpp
+++ b/src/hyteg/solvers/SORSmoother.hpp
@@ -40,6 +40,7 @@ class SORSmoother : public Solver< OperatorType >
                const typename OperatorType::dstType& b,
                const walberla::uint_t                level ) override
    {
+      fp_util::busy_sleep_for_millisecs( 1000 );
       if ( const auto* A_sor = dynamic_cast< const SORSmoothable< typename OperatorType::srcType >* >( &A ) )
       {
          A_sor->smooth_sor( x, b, relax_, level, flag_ );
